fe/fe-core/src/main/java/com/starrocks/analysis/RoutineLoadDataSourceProperties.java

```diff
@@ -62,7 +80,13 @@ public void analyze() throws AnalysisException {
     }
 
     public boolean hasAnalyzedProperties() {
-        return !kafkaPartitionOffsets.isEmpty() || !customKafkaProperties.isEmpty();
+        if (type == "KAFKA") {
+            return !kafkaPartitionOffsets.isEmpty() || !customKafkaProperties.isEmpty();
+        } else if (type.equals("PULSAR")) {
```

Comment:
Why does this use `type.equals`, while the line 83 use `type ==` ?

Reply:
`equals()` should be better, but I missed kafka code before, will fix it.